lib/css/components/select.less

```
@@ -0,0 +1,194 @@
+.s-select {
+    --_se-arrow-bc: currentColor transparent;
+    --_se-arrow-size: var(--su-static4); // Constant // TODO create name/styling convension for constants
+    --_se-select-bc: var(--bc-darker);
+    --_se-select-bc-focus: var(--theme-secondary-300);
+    --_se-select-bs-focus: 0 0 0 var(--su-static4) var(--focus-ring);
+    --_se-select-bg: var(--white);
+    --_se-select-br: var(--br-sm);
+    --_se-select-fc: var(--black);
+    --_se-select-px: 0.7em;
+    --_se-select-py: 0.6em;
+    --_se-select-fs: var(--fs-body1);
+
+    // CONTEXTUAL STYLES
+    @supports (-webkit-overflow-scrolling: touch) {
+        --_se-select-fs: 16px; // Increase font size for mobile safari. This keeps our inputs from zooming the page while focused
+        --_se-select-px: 0.55em; // Compensate for the larger font size so we generally keep the input the same size
+        --_se-select-py: 0.4em; // Compensate for the larger font size so we generally keep the input the same size
+    }
+
+    // MODIFIERS
+    .is-disabled &,
+    .is-readonly &,
+    .has-success &,
+    .has-error &,
+    .has-warning & {
+        position: relative;
+    }
+
+    .is-disabled & {
+        --_se-arrow-bc: var(--bc-darker) transparent;
+    }
+
+    &&__sm {
+        --_se-select-fs: var(--fs-caption);
+    }
+
+    &&__md {
+        --_se-select-br: calc(var(--br-sm) + var(--su-static1));
+        --_se-select-fs: var(--fs-body3);
+        --_se-select-py: 0.5em;
+    }
+
+    &&__lg {
+        --_se-select-br: calc(var(--br-sm) + var(--su-static1));
+        --_se-select-fs: var(--fs-title);
+        --_se-select-px: 0.6em;
+        --_se-select-py: 0.45em;
+    }
+
+    &&__xl {
+        --_se-select-br: var(--br-md);
+        --_se-select-fs: var(--fs-headline1);
+        --_se-select-px: 0.5em;
+        --_se-select-py: 0.4em;
+    }
+
+    .has-error & {
+        --_se-select-bc: var(--red-400);
+        --_se-select-bc-focus: var(--red-400);
+        --_se-select-bs-focus: 0 0 0 var(--su-static4) var(--focus-ring-error);
+    }
+
+    .has-success & {
+        --_se-select-bc: var(--green-400);
+        --_se-select-bc-focus: var(--green-400);
+        --_se-select-bs-focus: 0 0 0 var(--su-static4) var(--focus-ring-success);
+    }
+
+    .has-warning & {
+        --_se-select-bc: var(--yellow-600);
+        --_se-select-bc-focus: var(--yellow-600);
+        --_se-select-bs-focus: 0 0 0 var(--su-static4) var(--focus-ring-warning);
+    }
+
+    // CHILD ELEMENTS
+    &,
```

Added - double check visually. To make this more safe we should probably do:

```suggestion
    select&,
```